x-pack/plugins/security_solution/server/lists_integration/endpoint/validators/base_validator.test.ts

```diff
@@ -0,0 +1,187 @@
+  let exceptionLikeItem: ExceptionItemLikeOptions;
+  let validator: BaseValidatorMock;
+  let packagePolicyService: jest.Mocked<PackagePolicyServiceInterface>;
+  let initValidator: (withNoAuth?: boolean, withBasicLicense?: boolean) => BaseValidatorMock;
+
+  beforeEach(() => {
+    kibanaRequest = httpServerMock.createKibanaRequest();
+    exceptionLikeItem = createExceptionItemLikeOptionsMock();
+
+    const servicesStart = createMockEndpointAppContextServiceStartContract();
+
+    packagePolicyService =
+      servicesStart.packagePolicyService as jest.Mocked<PackagePolicyServiceInterface>;
+
+    endpointAppContextServices = new EndpointAppContextService();
+    endpointAppContextServices.setup(createMockEndpointAppContextServiceSetupContract());
+    endpointAppContextServices.start(servicesStart);
+
+    initValidator = (withNoAuth: boolean = false, withBasicLicense = false) => {
+      if (withNoAuth) {
+        const fleetAuthz = createFleetAuthzMock();
+        fleetAuthz.fleet.all = false;
+        (servicesStart.fleetAuthzService?.fromRequest as jest.Mock).mockResolvedValue(fleetAuthz);
+      }
+
+      if (withBasicLicense) {
+        (servicesStart.licenseService.isPlatinumPlus as jest.Mock).mockResolvedValue(false);
+      }
+
+      validator = new BaseValidatorMock(endpointAppContextServices, kibanaRequest);
+
+      return validator;
+    };
+  });
+
+  it('should use default endpoint authz (no access) when `request` is not provided', async () => {
+    const baseValidator = new BaseValidatorMock(endpointAppContextServices);
+
+    await expect(baseValidator._isAllowedToCreateArtifactsByPolicy()).resolves.toBe(false);
+    await expect(baseValidator._validateCanManageEndpointArtifacts()).rejects.toBeInstanceOf(
+      EndpointArtifactExceptionValidationError
+    );
+  });
+
+  it('should validate is allowed to manage endpoint artifacts', async () => {
+    await expect(initValidator()._validateCanManageEndpointArtifacts()).resolves.toBeUndefined();
+  });
+
+  it('should throw if not allowed to manage endpoint artifacts', async () => {
+    await expect(initValidator(true)._validateCanManageEndpointArtifacts()).rejects.toBeInstanceOf(
+      EndpointArtifactExceptionValidationError
+    );
+  });
+
+  it('should validate basic artifact data', async () => {
+    await expect(initValidator()._validateBasicData(exceptionLikeItem)).resolves.toBeUndefined();
+  });
+
+  it.each([
+    [
+      'name is empty',
+      () => {
+        exceptionLikeItem.name = '';
+      },
+    ],
+    [
+      'namespace is not agnostic',
+      () => {
+        exceptionLikeItem.namespaceType = 'single';
+      },
+    ],
+    [
+      'osTypes has more than 1 value',
+      () => {
+        exceptionLikeItem.osTypes = ['macos', 'linux'];
+      },
+    ],
+    [
+      'osType has invalid value',
+      () => {
+        exceptionLikeItem.osTypes = ['xunil' as 'linux'];
+      },
+    ],
+  ])('should throw if %s', async (_, setupData) => {
```

Comment:
fwiw, there is a way you can make these table tests look a bit better but potentially lose some typing by using a slightly different jest syntax I tried out the other day you can see [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/usage/detections/detection_rule_helpers.test.ts#L192-L225
):

<img width="1177" alt="Screen Shot 2022-01-19 at 11 24 51 AM" src="https://user-images.githubusercontent.com/1151048/150191469-818b7cff-8c77-44a9-ab24-e0be8099af49.png">

from:
https://jestjs.io/docs/api#testeachtablename-fn-timeout

Reply:
This is fine as-is, just mentioning the two pattern types.

Reply:
yeah, I know the `.each()` has a few formats you can use. I'll try out the tagged template approach next time. My assumption is that it will only work better if the variables/data are short and not that wide, else, content would span horizontally possibly requiring scrolling to see it.